### PR TITLE
fix: add validations into numberToBalance utils method

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -477,10 +479,47 @@ export function authorizationDataToAuthorization(auth: AuthorizationData): Autho
 /**
  * @hidden
  */
-export function numberToBalance(value: number | BigNumber, context: Context): Balance {
+export function numberToBalance(
+  value: number | BigNumber,
+  context: Context,
+  divisible?: boolean
```
Comment (monitz87) [inline src/utils/index.ts:485]: I know I said the default should be false, but I was mistaken. It should be true. Most amounts (such as POLYX amounts) have decimals, so divisibility is the sensible default

File: src/utils/__tests__/index.ts [hunk 2]
```diff
@@ -399,11 +404,63 @@ describe('numberToBalance and balanceToBigNumber', () => {
       .withArgs('Balance', value.multipliedBy(Math.pow(10, 6)).toString())
       .returns(fakeResult);
 
-    const result = numberToBalance(value, context);
+    let result = numberToBalance(value, context);
```
Comment (monitz87) [inline src/utils/__tests__/index.ts:407]: This test should explicity pass false

**Review by monitz87 — CHANGES_REQUESTED** [review]

**Review by monitz87 — CHANGES_REQUESTED** [review]
Also, `canSettle` should fetch the `isDivisible` value from the parent and pass it to the `numberToBalance`call

File: src/api/entities/SecurityToken/Settlements.ts [hunk 3]
```diff
@@ -39,6 +39,8 @@ export class Settlements extends Namespace<SecurityToken> {
 
     const { from = await this.context.getCurrentIdentity(), to, amount } = args;
 
+    const { isDivisible } = await details();
```
Comment (monitz87) [inline src/api/entities/SecurityToken/Settlements.ts:42]: You should group both these awaits into a `Promise.all`

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/entities/SecurityToken/Settlements.ts (hunk 3 above)
Comment (shuffledex) [reply src/api/entities/SecurityToken/Settlements.ts:42]: but we are going to make two request every time although `from` is set

Comment (monitz87) [reply src/api/entities/SecurityToken/Settlements.ts:42]:
```
const { to, amount } = args;
let { from } = args;
let isDivisible;

if (!from) {
  ([{ isDivisible }, from] = await Promise.all([details(), this.context.getCurrentIdentity()]);
} else {
  ({ isDivisible } = await details());
}
```

File: src/api/entities/SecurityToken/Settlements.ts [hunk 4]
```diff
@@ -37,9 +37,15 @@ export class Settlements extends Namespace<SecurityToken> {
...
     } = this;
 
-    const { from = await this.context.getCurrentIdentity(), to, amount } = args;
+    const { to, amount } = args;
+    let { from } = args;
+    let isDivisible;
 
-    const { isDivisible } = await details();
+    if (!from) {
+      [{ isDivisible }, from] = await Promise.all([details(), this.context.getCurrentIdentity()]);
```
Comment (monitz87) [inline src/api/entities/SecurityToken/Settlements.ts:45]:
```suggestion
      [{ isDivisible }, from] = await Promise.all([details(), context.getCurrentIdentity()]);
```

**Review by monitz87 — CHANGES_REQUESTED** [review]

**Review by monitz87 — APPROVED** [review]

**Pull request merged**